bat.d/41-framework

```diff
@@ -0,0 +1,536 @@
+        128) printf "* natacpi (%s) = inactive (no kernel support)\n" "$_batdrv_kmod" ;;
+        254) printf "* natacpi (%s) = inactive (laptop not supported)\n" "$_batdrv_kmod" ;;
+        *)   printf "* natacpi (%s) = unknown status\n" "$_batdrv_kmod" ;;
+    esac
+    # TEMPLATE: customize to vendor specs
+    if [ "$_bm_thresh" != "none" ]; then
+        printf "Parameter value ranges:\n"
+		printf "* START_CHARGE_THRESH_BAT1/1:  0 (not available)\n"
+        printf "* STOP_CHARGE_THRESH_BAT1/1:   0..100(default)\n"
+    fi
+    printf "\n"
+
+    # -- show battery data
+    local bat
+    local bcnt=0
+    local ed ef en
+    local efsum=0
+    local ensum=0
+
+    for bat in $_batteries; do # iterate batteries
+        batdrv_select_battery "$bat"
+
+        printf "+++ Battery Status: %s\n" "$bat"
+
+        printparm "%-59s = ##%s##" "$_bd_read/serial_number"
+
+        print_battery_cycle_count "$_bd_read/cycle_count" "$(read_sysf "$_bd_read/cycle_count")"
+
+        if [ -f "$_bd_read/energy_full" ]; then
+            printparm "%-59s = ##%6d## [mWh]" "$_bd_read/energy_full_design" "" 000
+            printparm "%-59s = ##%6d## [mWh]" "$_bd_read/energy_full" "" 000
+            printparm "%-59s = ##%6d## [mWh]" "$_bd_read/energy_now" "" 000
+            printparm "%-59s = ##%6d## [mW]" "$_bd_read/power_now" "" 000
+
+            # store values for charge / capacity calculation below
+            ed=$(read_sysval "$_bd_read/energy_full_design")
+            ef=$(read_sysval "$_bd_read/energy_full")
+            en=$(read_sysval "$_bd_read/energy_now")
+            efsum=$((efsum + ef))
+            ensum=$((ensum + en))
+
+        elif [ -f "$_bd_read/charge_full" ]; then
+            printparm "%-59s = ##%6d## [mAh]" "$_bd_read/charge_full_design" "" 000
+            printparm "%-59s = ##%6d## [mAh]" "$_bd_read/charge_full" "" 000
+            printparm "%-59s = ##%6d## [mAh]" "$_bd_read/charge_now" "" 000
+            printparm "%-59s = ##%6d## [mA]" "$_bd_read/current_now" "" 000
+
+            # store values for charge / capacity calculation below
+            ed=$(read_sysval "$_bd_read/charge_full_design")
+            ef=$(read_sysval "$_bd_read/charge_full")
+            en=$(read_sysval "$_bd_read/charge_now")
+            efsum=$((efsum + ef))
+            ensum=$((ensum + en))
+
+        else
+            ed=0
+            ef=0
+            en=0
+        fi
+
+        print_batstate "$_bd_read/status"
+        printf "\n"
+
+        if [ "$verbose" -eq 1 ]; then
+            printparm "%-59s = ##%6s## [mV]" "$_bd_read/voltage_min_design" "" 000
+            printparm "%-59s = ##%6s## [mV]" "$_bd_read/voltage_now" "" 000
+            printf "\n"
+        fi
+
+        # --- show battery features: thresholds
+        if [ "$_bm_thresh" = "natacpi" ]; then
+            printf "%-59s = %6s [%%]\n" "$_bf_stop"  "$(batdrv_read_threshold stop 1)"
```

Comment:
if you removed the first parameter in **batdrv_read_threshold()** above, also change to `$(batdrv_read_threshold 1)` here.